crown-court-contribution/src/test/java/uk/gov/justice/laa/crime/contribution/service/CompareContributionServiceTest.java

```diff
@@ -161,55 +126,21 @@ void givenActiveIdenticalContributionWithCaseTypeAndCorrespondenceStatusAppealCC
         ContributionResult contributionResult = TestModelDataBuilder.getContributionResult();
 
-        int result = compareContributionService.compareContribution(calculateContributionDTO, contributionResult);
-
-        assertThat(result).isOne();
-        verify(maatCourtDataService, times(1))
-                .updateCorrespondenceState(anyInt(), eq(CorrespondenceStatus.APPEAL_CC));
-    }
-
-    @Test
-    void givenActiveIdenticalContributionWithCorrespondenceStatusAppealCC_whenCompareContributionServiceIsInvoked_thenReturnTwo() {
         when(maatCourtDataService.findContribution(anyInt(), anyBoolean()))
                 .thenReturn(List.of(
                                 TestModelDataBuilder.buildContributionForCompareContributionService()
                         )
                 );
-        when(maatCourtDataService.findCorrespondenceState(anyInt())).
-                thenReturn(CorrespondenceStatus.APPEAL_CC);
         when(contributionService.hasMessageOutcomeChanged(anyString(), any()))
                 .thenReturn(false);
-        CalculateContributionDTO calculateContributionDTO =
-                TestModelDataBuilder.getContributionDTOForCompareContributionService(
-                        CaseType.COMMITAL.getCaseTypeString(),
-                        BigDecimal.valueOf(250),
-                        BigDecimal.valueOf(250),
-                        BigDecimal.valueOf(250),
-                        LocalDate.now(),
-                        MagCourtOutcome.APPEAL_TO_CC
-                );
-        ContributionResult contributionResult = TestModelDataBuilder.getContributionResult();
 
-        int result = compareContributionService.compareContribution(calculateContributionDTO, contributionResult);
+        boolean result = compareContributionService.shouldCreateContribution(calculateContributionDTO, contributionResult);
 
-        assertThat(result).isEqualTo(2);
-        verify(maatCourtDataService, times(1))
-                .updateCorrespondenceState(anyInt(), eq(CorrespondenceStatus.NONE));
+        assertThat(result).isTrue();
     }
 
     @Test
-    void givenActiveIdenticalContributionForCds15WorkAroundAndCorrespondenceStatusCds15_whenCompareContributionServiceIsInvoked_thenReturnTwo() {
-        when(maatCourtDataService.findContribution(anyInt(), anyBoolean()))
-                .thenReturn(List.of(
-                                TestModelDataBuilder.buildContributionForCompareContributionService()
-                        )
-                );
-        when(maatCourtDataService.findCorrespondenceState(anyInt()))
-                .thenReturn(CorrespondenceStatus.CDS15);
-        when(contributionService.hasMessageOutcomeChanged(anyString(), any()))
-                .thenReturn(false);
-        when(contributionService.isCds15WorkAround(any()))
-                .thenReturn(true);
+    void givenActiveIdenticalContributionNonAppealCCWithoutMagsOutcomeChange_whenShouldCreateContributionIsInvoked_thenReturnFalse() {
```

Comment:
Do we need one more test to cover when the contributions are identical, the mags outcome has not changed and the case type is appeal cc, to check that we also get back a result of `false`?

Reply:
There is a unit test for the above scenario that returns the result of True, just checking if the above scenarios you've written is the one you meant or if you meant to highlight a different scenario?

Reply:
I think we need another that checks for `false` for this block, specifically the OR on the case type (as you've tested everything else)

```
if (!activeContribution.isEmpty()
        && areContributionRecordsIdentical(contributionResult, contributions.get(0))
        && !(contributionService.hasMessageOutcomeChanged(magsCourtOutcome, repOrderDTO)
            || CaseType.APPEAL_CC.getCaseTypeString().equals(repOrderDTO.getCatyCaseType()))) {
    log.info("Contributions should not be created");
    return false;
}
```

Let me know if I'm mistaken on this though